PyMca5/PyMcaGui/io/TiledCatalogSelector.py

```diff
@@ -0,0 +1,145 @@
+import logging
+from collections import defaultdict
+from typing import Callable, List, Mapping, Optional, Sequence
+from urllib.parse import ParseResult, urlparse as _urlparse
+
+from PyQt5.QtCore import QEvent, QObject, pyqtSignal
+
+from tiled.client import from_uri
+from tiled.client.base import BaseClient
+
+
+_logger = logging.getLogger(__name__)
+
+
+class TiledCatalogSelectorSignals(QObject):
+    """Collection of signals for a TiledCatalogSelector model."""
+    client_connected = pyqtSignal(
+        str, # new URL
+        str, # new API URL
+        name="TiledCatalogSelector.client_connected",
+    )
+    
+    def __init__(self, parent: Optional[QObject] = None) -> None:
+        super().__init__(parent)
+
+
+class TiledCatalogSelector(object):
+    """View Model for selecting a Tiled CatalogOfBlueskyRuns."""
+    Signals = TiledCatalogSelectorSignals
+
+    def __init__(
+        self,
+        /,
+        url: str = None,
+        client: BaseClient = None,
+        validators: Mapping[str, List[Callable]] = None,
+        parent: Optional[QObject] = None,
+        *args,
+        **kwargs,
+    ):
+        _logger.debug("TiledCatalogSelector.__init__()...")
+
+        self.url = url
+        self.client = client
```

Comment:
Should initialize `self._url_buffer` here, rather than deal with run-time checks.

Reply:
Resolved in commit `07ea91fc`